packages/database/src/migrations/20210709031533-UnfpaMarshallIslandsMissing-modifies-data.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+import { codeToId, insertObject, generateId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const entityHierarchyName = 'unfpa';
+const projectEntityName = 'UNFPA';
+const countryEntityName = 'Marshall Islands';
+
+export const nameToId = async (db, table, name) => {
+  const record = await db.runSql(`SELECT id FROM "${table}" WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+exports.up = async function (db) {
+  const projectEntityId = await nameToId(db, 'entity', projectEntityName);
+  const countryEntityId = await nameToId(db, 'entity', countryEntityName);
+  const entityHierarchyId = await nameToId(db, 'entity_hierarchy', entityHierarchyName);
+  const facilities = (
+    await db.runSql(`
+    SELECT * from entity 
+    WHERE country_code = 'MH' AND "type" = 'facility'; 
+ `)
+  ).rows;
+
+  for (const facility of facilities) {
+    const { parent_id: ancestorId, id: descendantId } = facility;
+    // console.log(facility);
+    await insertObject(db, 'ancestor_descendant_relation', {
```

Comment:
Hmmm I think this might not be the right approach here. I think we actually automatically build up the `ancestor_descendant_relation` table via the `EntityHierarchyCacher`. It gets built from the `entity_relation` table, which is where we want to be adding the entities to. A good example to draw from would be: `tupaia/packages/database/src/migrations/20210615072057-UpdateUnfpaNonCanonicalRelations-modifies-data.js`

Reply:
eek - I think you are correct @rohan-bes looks like Marshall Islands got missed. And yes should be updating entity_relation not ancestor_descendant_relation.

Reply:
Not your fault @nick-beyondessential I think it's a very easy mistake to make. The canonical hierarchy for UNFPA goes country->district->facility, which is the same hierarchy that Marshall Islands uses. However, since the relations at generation distance 2 have some entity_relations defined for other countries, it stop using the canonical hierarchy for that generation. I think we should be pretty wary of how we talk about relations within hierarchies, since often we talk about type->type relations (eg. country->district) but the caching logic just relations on distance relations...

Personally I'd love if we just made the call to do away with canonical hierarchies all together, and had to have explicit entity_relations defined for everything... I know it would blow up the table size, but the hybrid model confuses me quite often!